src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ExportInterceptingPropertyValueProvider]
+    internal sealed class TargetFrameworkValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+
+        [ImportingConstructor]
+        public TargetFrameworkValueProvider(IUnconfiguredProjectVsServices projectVsServices)
+        {
+            Requires.NotNull(projectVsServices, nameof(projectVsServices));
+
+            _projectVsServices = projectVsServices;
+        }
+
+        public override string GetPropertyName() => "TargetFramework";
+
+        public override async Task<string> OnGetEvaluatedPropertyValueAsync(string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            Requires.NotNull(_projectVsServices.VsHierarchy, "vsHierarchy");
+
+            // Fetch the target framework version from the VSHierarchy.
+            object targetFrameworkVersion;
+            if (ErrorHandler.Succeeded(_projectVsServices.VsHierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)Shell.VsHierarchyPropID.TargetFrameworkVersion, out targetFrameworkVersion)))
+            {
+                return ((uint)targetFrameworkVersion).ToString();
```

Comment:
Adding it to ConfigurationGeneral will ensure it shows up on that type, but how will it be implemented? There is no MSBuild property named "TargetFramework" in the project file.

TargetFramework was already non-visible a string property defined in [GeneralBrowseObject.xaml](https://github.com/dotnet/roslyn-project-system/pull/287/files#diff-a7ea57aece378dbf2944a9c60361a729L29). I had to change it to IntProperty as SigningPropPage (and potentially other consumers) expect this to be integral.